### PR TITLE
MIG-549: MigHook doc improvement

### PR DESCRIPTION
**Pull request by apinnick**
https://issues.redhat.com/browse/MIG-549

Updated sections:
https://deploy-preview-29833--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_1_4/migrating-application-workloads-4-1-4.html#migration-about-migration-hooks_migrating-4-1-4

https://deploy-preview-29833--osdocs.netlify.app/openshift-enterprise/latest/migration/migrating_4_1_4/migrating-applications-with-cam-4-1-4.html#migration-writing-ansible-playbook-hook_migrating-4-1-4

### CODE REVIEW

**Comment by XinRedhat** [comment]
LGTM

**Review by pneedle-rh — COMMENTED** [review]
@apinnick I have now completed my review. This is a great addition! Please let me know if you have any questions about my suggestions. Thanks!

File: modules/migration-about-migration-hooks.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+The Ansible playbook is mounted on a hook container as a config map. The hook container runs as a job, using the cluster, service account, and namespace specified in the `MigPlan` CR. The job continues to run until it reaches the default limit for retries, `6`, or a successful completion, even if the initial pod is evicted or killed.
+
+The default Ansible runtime image is `registry.redhat.io/rhmtc/openshift-migration-hook-runner-rhel7:latest`. This image is based on the Ansible Runner image and includes `python-openshift` for Ansible Kubernetes resources and an updated `oc` binary.
+
+Optional: You can use a custom Ansible runtime image, containing additional Ansible modules or tools, instead of the default image.
+
+.Custom hook container
+
+You can create a custom hook container that includes Ansible playbooks or a different programming language.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/migration-about-migration-hooks.adoc:32]: You're right but what else do we call "any compute program the user wants"?

File: modules/migration-creating-migration-plan-cam.adoc [hunk 2]
```diff
@@ -67,23 +67,28 @@ You can change the storage class of data migrated with *Filesystem copy*.
...
+.. Optional: Specify the Ansible runtime image if you are not using the default hook image, `quay.io/konveyor/hook-runner:latest`.
+.. If the hook is not an Ansible playbook, select *Custom container image* and specify the image name and path.
++
+A custom container image can contain Ansible playbook or it can be written in a different programming language.
+
+.. Select *Source cluster* or *Target cluster*.
+.. Enter the *Service account name* and the *Service account namespace*.
+.. Select the migration step for the hook:
+
+* *preBackup*: Before the application workload is backed up on the source cluster
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/migration-creating-migration-plan-cam.adoc:85]: Here, they are part of the UI, so they are bold. In the migration-about-migration-hooks module, they are names of variables.

File: modules/migration-mtc-cr-manifests.adoc [hunk 3]
```diff
@@ -287,32 +287,35 @@ spec:
...
-<8> Direct volume migration is disabled if `true`. PVs are copied from the source cluster to the replication repository and from the replication repository to the destination cluster.
-<9> Specify the name of `MigStorage` CR.
-<10> Specify one or more namespaces.
-<11> The `MigPlan` CR is validated if `true`.
+<5> Optional: Specify the namespace in which the hook will run.
+<6> Optional: Specify the migration phase during which a hook runs. One hook can be assigned to one phase. The expected values are `PreBackup`, `PostBackup`, `PreRestore`, and `PostRestore`.
+<7> Optional: Specify the name of the `MigHook` CR.
+<8> Optional: Specify the namespace of `MigHook` CR.
+<9> Optional: Specify a service account with `cluster-admin` privileges.
+<10> Direct image migration is disabled if `true`. Images are copied from the source cluster to the replication repository and from the replication repository to the destination cluster.
```
[the remark this answers is not in the document]
Comment (apinnick) [reply modules/migration-mtc-cr-manifests.adoc:314]: Not necessary.

File: migration/migrating_3_4/migrating-applications-with-cam-3-4.adoc [hunk 4]
```diff
@@ -8,9 +8,16 @@ toc::[]
 
 You can migrate your applications by using the {mtc-full} ({mtc-short}) web console or on the command line.
 
-If you are using self-signed CA certificates to secure the clusters or the replication repository, you can create a CA certificate bundle file or disable SSL verification.
+include::modules/migration-prerequisites.adoc[leveloffset=+1]
+include::modules/migration-creating-ca-bundle.adoc[leveloffset=+2]
+include::modules/migration-writing-ansible-playbook-hook.adoc[leveloffset=+2]
 
-include::modules/migration-creating-ca-bundle.adoc[leveloffset=+1]
+[id="migrating-applications-additional-resources_{context}"]
```
[the remark this answers is not in the document]
Comment (apinnick) [reply migration/migrating_3_4/migrating-applications-with-cam-3-4.adoc:15]: Yes

**Pull request merged**